packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx

```diff
@@ -0,0 +1,1130 @@
+   * @param index - the index to get the next selectable index from
+   * @param delta - optional delta to step by when finding the next index, defaults to 0
+   * @returns {number} - the next valid selectable index. If the new index is outside of the bounds,
+   * it will snap to the edge of the options array. If delta == 0 and the given index is not selectable
+   */
+  private _getNextSelectableIndex(index: number, searchDirection: SearchDirection): number {
+    let { currentOptions } = this.state;
+
+    let newIndex = index + searchDirection;
+
+    newIndex = Math.max(0, Math.min(currentOptions.length - 1, newIndex));
+
+    let option: IComboBoxOption = currentOptions[newIndex];
+
+    // attempt to skip headers and dividers
+    if ((option.itemType === DropdownMenuItemType.Header ||
+      option.itemType === DropdownMenuItemType.Divider)) {
+
+      // Should we continue looking for an index to select?
+      if (searchDirection !== SearchDirection.none &&
+        ((newIndex !== 0 && searchDirection < SearchDirection.none) ||
+          (newIndex !== currentOptions.length - 1 && searchDirection > SearchDirection.none))) {
+        newIndex = this._getNextSelectableIndex(newIndex, searchDirection);
+      } else {
+        // If we cannot perform a useful search just return the index we were given
+        return index;
+      }
+    }
+
+    // We have the next valid selectable index, return it
+    return newIndex;
+  }
+
+  /**
+   * Set the selected index. Note, this is
+   * the "real" selected index, not the pending selected index
+   * @param index - the index to set (or the index to set from if a search direction is provided)
+   * @param searchDirection - the direction to search along the options from the given index
+   */
+  private _setSelectedIndex(index: number, searchDirection: SearchDirection = SearchDirection.none) {
+    let { onChanged } = this.props;
+    let { selectedIndex, currentOptions } = this.state;
+
+    // Find the next selectable index, if searchDirection is none
+    // we will get our starting index back
+    index = this._getNextSelectableIndex(index, searchDirection);
+
+    // Are we at a new index? If so, update the state, otherwise
+    // there is nothing to do
+    if (index !== selectedIndex) {
+      let option: IComboBoxOption = currentOptions[index];
+
+      // Set the selected option
+      this.setState({
+        selectedIndex: index
+      });
+
+      // Did the creator give us an onChanged callback?
+      if (onChanged) {
+        onChanged(option, index);
+      }
+
+      // if we have a new selected index,
+      // clear all of the pending info
+      this._clearPendingInfo();
+    }
+  }
+
+  /**
+   * Focus (and select) the content of the input
+   * and set the focused state
+   */
+  @autobind
+  private _select() {
+    this._comboBox.inputElement.select();
+
+    if (!this.state.focused) {
+      this.setState({ focused: true });
```

Comment:
The local state has to do with styling the whole comboBox to give it the "focused" look. While we are "dealing with" a comboBox, it should continue to have the "focused" look